### PR TITLE
gnrc_sixlowpan_nd_router: initial import of router behavior of 6LoWPAN-ND

### PR DESCRIPTION
**Pull request by authmillenon**
This imports [router behavior](https://tools.ietf.org/html/rfc6775#section-6) for [6LoWPAN-ND](https://tools.ietf.org/html/rfc6775) and adapts GNRC accordingly

Depends on:
* ~~#3049: router-behavior for normal router discovery~~ (merged)
* ~~#3628: address-less link-layer handling for NDP~~ (merged)
* ~~#3744: `gnrc_ipv6_nc` adaption for 6LoWPAN-ND~~ (merged)
* ~~#3745: `gnrc_ipv6_netif` adaptations for router discovery~~ (merged)
* ~~#3746: host-behavior for normal router~~ (merged)
* ~~#3747: adaption of the NA send function to build ARO externally~~ (merged)
* ~~#3748: host-behavior for 6LoWPAN-ND~~ (merged)

### CODE REVIEW

**Comment by authmillenon** [comment]
Rebased to current master and dependencies

**Comment by authmillenon** [comment]
Rebased to current master and dependenies

**Comment by authmillenon** [comment]
Rebased to current #3748.

**Comment by authmillenon** [comment]
Rebased to current #3748

**Comment by authmillenon** [comment]
Rebased to current #3748 and reverted the temporary fix for configuration of solicited-nodes multicast addresses.

**Comment by authmillenon** [comment]
Rebased to current master

**Comment by authmillenon** [comment]
Rebased to current master

File: sys/include/net/gnrc/sixlowpan/nd/router.h [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ */
+static inline void gnrc_sixlowpan_nd_router_set_rtr_adv(gnrc_ipv6_netif_t *netif, bool enable)
+{
+    if (enable) {
+        netif->flags |= GNRC_IPV6_NETIF_FLAGS_RTR_ADV;
+    }
+    else {
+        netif->flags &= ~GNRC_IPV6_NETIF_FLAGS_RTR_ADV;
+    }
+}
```
Comment (OlegHahm) [inline sys/include/net/gnrc/sixlowpan/nd/router.h]: These setter functions look much like Java code to me.

File: sys/net/gnrc/network_layer/ndp/gnrc_ndp.c [hunk 2]
```diff
@@ -52,7 +53,19 @@ static void _stale_nc(kernel_pid_t iface, ipv6_addr_t *ipaddr, uint8_t *l2addr,
 {
     if (l2addr_len != -ENOTSUP) {
         gnrc_ipv6_nc_t *nc_entry = gnrc_ipv6_nc_get(iface, ipaddr);
-        if (nc_entry == NULL) {
+        if (nc_entry == NULL) {            /* tentative type see https://tools.ietf.org/html/rfc6775#section-6.3*/
```
Comment (OlegHahm) [inline sys/net/gnrc/network_layer/ndp/gnrc_ndp.c]: line seems too long

File: sys/include/net/gnrc/sixlowpan/nd/router.h (hunk 1 above)
Comment (authmillenon) [inline sys/include/net/gnrc/sixlowpan/nd/router.h]: Bit-operations are not typical java operations. Moreover, it makes https://github.com/RIOT-OS/RIOT/pull/3749/files#diff-f9bf85cb337616822921e4230ede1572R239 more readable.

Comment (OlegHahm) [inline sys/include/net/gnrc/sixlowpan/nd/router.h]: I don't think that anything gets more readable by replacing an assignment by a function and IMO it adds just 18 lines of additional code plus documentation, but this is not a blocker for this PR.

File: sys/net/gnrc/network_layer/ndp/gnrc_ndp.c (hunk 2 above)
Comment (authmillenon) [inline sys/net/gnrc/network_layer/ndp/gnrc_ndp.c]: Fixed

**Comment by authmillenon** [comment]
Fixed doc issue

**Comment by OlegHahm** [comment]
Not really related to this PR, but fixing parsing of router advertisements in general: https://github.com/authmillenon/RIOT/pull/20

**Comment by OlegHahm** [comment]
ACK, please squash.

**Comment by authmillenon** [comment]
Squashed

**Pull request merged**